### PR TITLE
Add Data Explorer InfluxQL query and location query synchronization

### PR DESCRIPTION
**Pull request by cryptoquick**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect #1815 

### The problem
There's been a need to share queries in the data explorer using a URL.

### The Solution
Add Data Explorer InfluxQL query and location query synchronization, so DE queries can be shared via a a URL.

### CODE REVIEW

File: ui/src/data_explorer/containers/DataExplorer.js [hunk 1]
```diff
@@ -35,6 +38,25 @@ class DataExplorer extends Component {
     return queryConfigs[0]
   }
 
+  componentDidMount() {
+    const {query} = queryString.parse(window.location.search)
+    if (query && query.length) {
```
Comment (121watts) [inline ui/src/data_explorer/containers/DataExplorer.js]: Let's limit the amount of times we use the window object.  Let's use `router.location.search` here instead.

File: ui/src/data_explorer/containers/DataExplorer.js [hunk 2]
```diff
@@ -35,6 +38,25 @@ class DataExplorer extends Component {
...
+
+  componentWillReceiveProps(nextProps) {
+    const {router} = this.props
+    const {queryConfigs, timeRange} = nextProps
+    const query = buildRawText(queryConfigs[0], timeRange)
+    const qsCurrent = queryString.parse(router.location.search)
+    if (query.length && qsCurrent.query !== query) {
+      const qsNew = queryString.stringify({query})
+      router.push(`${window.location.pathname}?${qsNew}`)
+    }
```
Comment (121watts) [inline ui/src/data_explorer/containers/DataExplorer.js]: `router.location.pathname` should work here as well

File: ui/src/data_explorer/containers/DataExplorer.js [hunk 3]
```diff
@@ -35,6 +38,25 @@ class DataExplorer extends Component {
...
+      const qc = this.props.queryConfigs[0]
+      this.props.queryConfigActions.editRawText(qc.id, query)
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {router} = this.props
+    const {queryConfigs, timeRange} = nextProps
+    const query = buildRawText(queryConfigs[0], timeRange)
+    const qsCurrent = queryString.parse(router.location.search)
```
Comment (121watts) [inline ui/src/data_explorer/containers/DataExplorer.js]: Indexing on queryConfigs here seems like its begging for a runtime error.  Maybe guard against this with a `_.get()`?

File: ui/src/data_explorer/containers/DataExplorer.js (hunk 1 above)
Comment (cryptoquick) [reply ui/src/data_explorer/containers/DataExplorer.js]: oh right! my oversight 😸

File: ui/src/data_explorer/containers/DataExplorer.js (hunk 3 above)
Comment (cryptoquick) [reply ui/src/data_explorer/containers/DataExplorer.js]: sounds good

File: ui/src/data_explorer/containers/DataExplorer.js (hunk 2 above)
Comment (cryptoquick) [reply ui/src/data_explorer/containers/DataExplorer.js]: yep!

**Pull request merged**